tests/test_example_dags.py

```diff
@@ -0,0 +1,41 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import unittest
+from glob import glob
+
+from airflow.models import DagBag
+
```

Comment:
For new tests, did you consider using `pytest` instead of `unittest`? This test can be a simple function with `pytest`. With `pytest.mark.paraemtrize`, individual modules can be created as sub tests automatically. Something like this will do. Having the relative_filepath in the test name (instead of full path) is just to make sure test names don't change.

One other suggestion you may consider is to mock `log.exception()` to raise error (instead of counting `import_errors`). That way, when a DAG fails to import and `log.exception()` is called inside dagbag.py, the stack trace is printed immediately by pytest instead of printing the exception to the terminal output together with other output.

```python
import os
import pytest

@pytest.mark.parametrize("relative_filepath", [os.realpath(p, ROOT_FOLDER) 
                                               for p in glob(f"{ROOT_FOLDER}/airflow/**/example_dags/example_*.py", 
                                                             recursive=True))])
def test_should_be_importable(relative_filepath):
    dagbag = DagBag(
        dag_folder=os.path.join(ROOT_FOLDER, relative_filepath),
        executor=None,
        include_examples=False,
    )
    assert len(dagbag.import_errors) == 0, f"import_errors={str(dagbag.import_errors)}" 
    assert len(dagbag.dag_ids) >= 1, f"{relative_filepath} should contain dags"
```

Reply:
We had a long discussion about pytest and parameterized. https://lists.apache.org/thread.html/1e4df7d4b0cd9b2d2bb76a3336471aa85e852545dd41ada6d4e461b8%40%3Cdev.airflow.apache.org%3E

The use of subtest is more nicely displayed in the IDE.
![Screenshot 2020-02-17 at 20 38 16](https://user-images.githubusercontent.com/12058428/74682183-a06b1180-51c5-11ea-98e1-7582353450de.png)
Using parameterized will fake us the number of tests, because suddenly the number of tests will increase by 80. If it's really not a few tests, but only one. I also don't like parameterized when generating a list is expensive. Here it is necessary to walk through a large number of directories to create these tests. 

I don't see the benefit of changing the place where this message is to be displayed. On the other hand, the lack of mock makes this test easier to understand.